src/NuGet.Clients/NuGet.PackageManagement.UI/Options/PackageSourcesOptionsControl.cs

```diff
@@ -661,6 +661,23 @@ private static bool IsPathRootedSafe(string path)
             // Otherwise, Path.IsPathRooted() will throw an ArgumentException.
             return path.IndexOfAny(Path.GetInvalidPathChars()) == -1 && Path.IsPathRooted(path);
         }
+
+        private void NewPackageSource_TextChanged(object sender, EventArgs e)
+        {
+            var source = new PackageSource(NewPackageSource.Text.Trim());
```

Comment:
This doesn't feel complete to me. The package source options has two text boxes, one for human friendly name, the other for the URL, but only one of these is set in the new instance of `PackageSource`.  Is the other missing?

In `PackageSourceCheckedListBox.cs` you have `new PackageSource(currentItem.Source, currentItem.Name)`